test/DurableTask.SqlServer.AzureFunctions.Tests/MultiTenancyCoreScenarios.cs

```diff
@@ -0,0 +1,41 @@
+namespace DurableTask.SqlServer.AzureFunctions.Tests
+{
+    using System;
+    using System.Threading.Tasks;
+    using Microsoft.Azure.WebJobs.Extensions.DurableTask;
+    using Xunit;
+    using Xunit.Abstractions;
+
+    public class MultiTenancyCoreScenarios : CoreScenarios
+    {
+        public MultiTenancyCoreScenarios(ITestOutputHelper output)
+            : base(output, "TaskHubWithMultiTenancy", true)
+        {
+        }
+
+        [Fact]
+        public async Task When_WithMultiTenancy_should_Not_Take_Care_Of_TaskHub()
+        {
+            string otherTaskHubName = "SomeOtherTaskHub";
+
+            string currentTaskHubInstanceId = Guid.NewGuid().ToString();
+            await this.StartOrchestrationWithoutWaitingAsync(nameof(Functions.Sequence), instanceId: currentTaskHubInstanceId);
+
+            string anotherTaskHubInstanceId = Guid.NewGuid().ToString();
+            await this.StartOrchestrationWithoutWaitingAsync(nameof(Functions.Sequence), instanceId: anotherTaskHubInstanceId, taskHub: otherTaskHubName);
+
+            IDurableClient client = this.GetDurableClient();
+            var current = await client.GetStatusAsync(currentTaskHubInstanceId);
+            Assert.NotNull(current);
+            var otherInstance = await client.GetStatusAsync(anotherTaskHubInstanceId);
+            Assert.NotNull(otherInstance);
+
+            IDurableClient clientOtherTaskHub = this.GetDurableClient(otherTaskHubName);
+            var currentFromOtherTaskHub = await clientOtherTaskHub.GetStatusAsync(currentTaskHubInstanceId);
+            Assert.NotNull(currentFromOtherTaskHub);
```

Comment:
This is confusing to me. If you have two different task hubs, why should a client configured for one be able to fetch status for the other?

Reply:
It should not but it s the case and that what leaded to the bug for me. Maybe instead we should throw if a custom taskhub is used when in multitenancy mode

Reply:
I removed it but I think there should be an issue about this.